File: brute_force/programmers/practice_kit_2.py
import itertools 

def solution(numbers):
    answer = 0
    num = []
    
    for i in range(1, len(numbers) + 1):
        # n 개에서 i개 뽑을 때의 순열 
        # permutation 객체 반환. 튜플들로 구성된 객체 
        pm = itertools.permutations(numbers, i)
        # 각 경우가 튜플로 되어있으므로 합치고, int로 바꾸기
        for n in map(int, map(''.join, pm)):
            num.append(n)

    # 숫자가 같을 경우가 있으면 중복 발생하므로 중복제거
    num = set(num)
    # 소수 아닌 경우 카운트
    for n in num:
        if n <= 1:
            answer += 1
            continue

        for i in range(2, n):
            if n % i == 0:
                answer += 1
                break

    return len(num) - answer

File: brute_force/programmers/Practice_kit_2.java
package brute_force.programmers;

import java.util.ArrayList;

public class Practice_kit_2 {
    static int count = 0;
    static ArrayList<Integer> numArray = new ArrayList<>();

    public static void main(String[] args) {
        String numbers = "011";
        System.out.println(solution(numbers));
    }

    static int solution(String numbers) {
        int n = numbers.length();
        char[] arr = new char[n];
        boolean[] visited = new boolean[n];

        for (int i = 0; i < n; i++) {
            arr[i] = numbers.charAt(i);
        }

        for (int i = 1; i <= n ; i++) {
            dfs(i, 0, arr, visited, new char[i]);
        }

        for (int num : numArray) {
            isPrime(num);
        }
 
        return count;
    }

    static void isPrime(int num) {
        if (num < 2) return;

        for (int i = 2; i < num; i++) {
            if (num % i == 0) return;
        }

        count++;
    }

    static void dfs(int m, int index, char[] arr, boolean[] visited, char[] res) {
        if (m == index) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < res.length; i++) {
                sb.append(res[i]);
            }
            
            int num = Integer.parseInt(sb.toString());
            if (!numArray.contains(num)) {
                numArray.add(num);
            }
        
            return;
        }

        for (int i = 0; i < arr.length; i++) {
            if (!visited[i]) {
                visited[i] = true;
                res[index] = arr[i];
                dfs(m, index + 1, arr, visited, res);
                visited[i] = false;
            }
        }
    }
}
